### PR TITLE
 Laboratory work #3, Alyona Gosteva - 21FPL2

### PR DESCRIPTION
**Pull request by Gostaa**

### CODE REVIEW

**Comment by artyomtugaryov** [comment]
Looks like the PR contains implementations of two labs - 2 and 3. You have to split it to 2 PRs - one for the 2 and another for the 3 lab

File: lab_2_keywords_cooccurrence/main.py [hunk 1]
```diff
@@ -48,13 +51,33 @@ def extract_candidate_keyword_phrases(phrases: Sequence[str], stop_words: Sequen
...
+
+    for phrase in phrases:
+        tokens_list.append(phrase.lower().split())
+
+    for tokens in tokens_list:
+        new_candidate = []
+        for token in tokens:
+            if token not in stop_words:
+                new_candidate.append(token)
+            else:
```
Comment (artyomtugaryov) [inline lab_2_keywords_cooccurrence/main.py:74]: elif

File: lab_2_keywords_cooccurrence/main.py [hunk 2]
```diff
@@ -83,12 +120,25 @@ def calculate_word_degrees(candidate_keyword_phrases: KeyPhrases,
...
-        word_degrees[token] = sum(len(phrase) for phrase in candidate_keyword_phrases if token in phrase)
-    return word_degrees
+    if not check_list_types(candidate_keyword_phrases, tuple) or not check_list_types(content_words, str):
+        return None
+    for sequence in candidate_keyword_phrases:
+        if not check_list_types(sequence, str):
+            return None
+
+    word_degrees_dict = {}
+    for word in content_words:
```
Comment (artyomtugaryov) [inline lab_2_keywords_cooccurrence/main.py:134]: I believe you can integrate this for to the for of counting `word_degrees_dict`

File: lab_3_keywords_textrank/main.py [hunk 3]
```diff
@@ -25,6 +25,7 @@ class TextPreprocessor:
     preprocess_text(text: str) -> tuple[str, ...]:
         Produces filtered clean lowercase tokens from raw text
     """
+
```
Comment (artyomtugaryov) [inline lab_3_keywords_textrank/main.py:28]: revert

File: lab_3_keywords_textrank/main.py [hunk 4]
```diff
@@ -51,7 +53,10 @@ def _clean_and_tokenize(self, text: str) -> tuple[str, ...]:
             tuple[str, ...]
                 clean lowercase tokens
         """
-        pass
+        for symbol in self._punctuation:
+            text = text.replace(symbol, '')
+        tokens = tuple(text.lower().split())
```
Comment (artyomtugaryov) [inline lab_3_keywords_textrank/main.py:58]: return `tuple(text.lower().split())` directly

File: lab_3_keywords_textrank/main.py [hunk 5]
```diff
@@ -66,7 +71,8 @@ def _remove_stop_words(self, tokens: tuple[str, ...]) -> tuple[str, ...]:
             tuple[str, ...]
                 tokens without stop-words
         """
-        pass
+        filtered_tokens = tuple(token for token in tokens if token not in self._stop_words)
```
Comment (artyomtugaryov) [inline lab_3_keywords_textrank/main.py:74]: again

File: lab_3_keywords_textrank/main.py [hunk 6]
```diff
@@ -136,7 +149,12 @@ def encode(self, tokens: tuple[str, ...]) -> Optional[tuple[int, ...]]:
                 sequence of integer tokens
         In case of empty tokens input data, None is returned
         """
-        pass
+        if not tokens:
+            return None
+
+        self._learn_indices(tokens)
+        encoded_tokens = tuple(self._word2id[token] for token in tokens)
```
Comment (artyomtugaryov) [inline lab_3_keywords_textrank/main.py:156]: again - you do not need create the `encoded_tokens` variable

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_2_keywords_cooccurrence/main.py (hunk 1 above)
Comment (Gostaa) [reply lab_2_keywords_cooccurrence/main.py:74]: done

File: lab_3_keywords_textrank/main.py (hunk 3 above)
Comment (Gostaa) [reply lab_3_keywords_textrank/main.py:28]: done

File: lab_3_keywords_textrank/main.py (hunk 4 above)
Comment (Gostaa) [reply lab_3_keywords_textrank/main.py:58]: done

File: lab_3_keywords_textrank/main.py (hunk 5 above)
Comment (Gostaa) [reply lab_3_keywords_textrank/main.py:74]: done

File: lab_3_keywords_textrank/main.py (hunk 6 above)
Comment (Gostaa) [reply lab_3_keywords_textrank/main.py:156]: done

File: lab_2_keywords_cooccurrence/main.py (hunk 2 above)
Comment (Gostaa) [reply lab_2_keywords_cooccurrence/main.py:133]: done

**Pull request closed without merging**